dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/theme/ThemeResourceIntegrationTest.java

```diff
@@ -50,120 +64,224 @@
     }
 
+    /**
+     * Method to test: findThemes
+     * Given Scenario: Find all the themes of the system host.
+     * ExpectedResult: The endpoint should return 200, system theme should be returned
+     *
+     */
     @Test
-    public void test_FindThemes_WhenHostIdIsSentAndPerPageEquals1_Returns1Theme()
-            throws Throwable {
-        final ThemeResource resource = new ThemeResource();
+    public void test_FindThemes_WhenHostIdIsSystemTheme_ReturnsSystemTheme() {
 
-        final Response response = resource.findThemes(getHttpRequest(),  new EmptyHttpResponse(), host.getIdentifier(), 0, 1,
+        final Response responseResource = resource.findThemes(getHttpRequest(null),  response, APILocator.systemHost().getIdentifier(), 0, -1,
                 OrderDirection.ASC.name(), null);
 
-        validateFindThemesResponse(response, 1);
+        assertEquals(Status.OK.getStatusCode(), responseResource.getStatus());
+        final ResponseEntityView responseEntityView = ResponseEntityView.class.cast(responseResource.getEntity());
+        final PaginatedArrayList themeList = PaginatedArrayList.class.cast(responseEntityView.getEntity());
+        assertEquals(1,themeList.size());
+        assertEquals(APILocator.getThemeAPI().systemTheme().getIdentifier(),HashMap.class.cast(themeList.get(0)).get("identifier"));
     }
 
-    @Test
-    public void testFindThemes_PaginationTotalEntries_ShouldMatch_totalCountPerHost()
-            throws Throwable {
-        final ThemeResource resource = new ThemeResource();
+    /**
+     * Method to test: findThemes
+     * Given Scenario: Tries to find the themes without sending the hostId (required field).
+     * ExpectedResult: The endpoint should return BadRequest
+     *
+     */
+    @Test(expected = BadRequestException.class)
+    public void test_FindThemes_WhenHostIdIsNotSent_ReturnsBadRequest() {
+        resource.findThemes(getHttpRequest(null),  response, "", 0, -1,
+                OrderDirection.ASC.name(), null);
+    }
 
-        final int expectedTotal = getTotalCount(resource, getHttpRequest(), host.getIdentifier(), -1);
+    /**
+     * Method to test: findThemes
+     * Given Scenario: Tries to find the themes but the hostId provided does not belong to any Host.
+     * ExpectedResult: The endpoint should return DotDataException 404
+     *
+     */
+    @Test(expected = DoesNotExistException.class)
+    public void test_FindThemes_WhenInvalidHostId_ReturnsDoesNotExistException() {
+        resource.findThemes(getHttpRequest(null),  response, "Id-Not-Exist", 0, -1,
+                OrderDirection.ASC.name(), null);
+    }
 
-        final Response response = resource.findThemes(getHttpRequest(),  new EmptyHttpResponse(), host.getIdentifier(), 0, 1,
+    /**
+     * Method to test: findThemes
+     * Given Scenario: Find all the themes of a specific host, sending the host id as a filter but as a limited user.
+     * ExpectedResult: DotSecurityException
```

Comment:
Here is says expected `DotSecurityException` but the test actual expects `PaginationException` which might not translate into the proper response code. Please check @erickgonzalez

Reply:
updated to PaginationException since DotSecurityException is caught by the Paginator and throws PaginationException